.github/workflows/run-test-cases.yml

```diff
@@ -58,25 +59,88 @@ on:
       - published
     
 jobs:
+  build-containers:
+    # Build containers only if this is a PR ... otherwise, the containers are pulled ghcr
+    if: >-
+      ( github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.fork == true ) ||
+      ( github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == false )
+    runs-on: ubuntu-18.04
+    timeout-minutes: 35
+
+    steps:
+    - name: Checkout the merged commit from PR and base branch
+      uses: actions/checkout@v2
+      if: github.event_name == 'pull_request_target'
+      with:
+        # pull_request_target is run in the context of the base repository
+        # of the pull request, so the default ref is master branch and
+        # ref should be manually set to the head of the PR
+        ref: refs/pull/${{ github.event.pull_request.number }}/head
+
+    - name: Checkout the head commit of the branch
+      if: ${{ github.event_name != 'pull_request_target' }}
+      uses: actions/checkout@v2
+
+    - name: Log into dockerhub to avoid throttled anonymous dockerhub pulls
+      run: echo "${{ secrets.DHPASSWORD }}" | docker login --username "${{ secrets.DHUSERNAME }}" --password-stdin
+
+    - name: Log into ghcr to access intermediate build containers
+      if: startsWith(github.event_name, 'pull_request')
+      run: echo "${{ secrets.crPassword }}" | docker login --username "${{ secrets.crUsername }}" ghcr.io --password-stdin
+
+    - name: Build local containers for PR tests
+      if: startsWith(github.event_name, 'pull_request')
+      env:
+        BUILD_AMD64: 1
+        BUILD_ARM32: 0
+        BUILD_ARM64: 0 
+        PREFIX: ghcr.io/deislabs/akri
+        LABEL_PREFIX: pr
+      run: |
+        make akri-build
+        make controller-build-amd64
+        make agent-build-amd64
+        docker save ${PREFIX}/agent:${LABEL_PREFIX}-amd64 > agent.tar
+        docker save ${PREFIX}/controller:${LABEL_PREFIX}-amd64 > controller.tar
```

Comment:
Should we add logout?

Reply:
i'm not clear what you have in mind?  do you mean something like:

echo "Saving ${PREFIX}/agent:${LABEL_PREFIX}-amd64 to agent.tar"
docker save ${PREFIX}/agent:${LABEL_PREFIX}-amd64 > agent.tar

Reply:
this isn't intended to fix the credential thing.  is that what you are referring to?

Reply:
Yes. If you think it would not help anything, fine.

Reply:
ok cool ... i agree we need to fix that, but i think it would be better to address that more holistically than here.